tools/specs/lib/upgradeTilesetSpec.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+var fsExtra = require('fs-extra');
+var GltfPipeline = require('gltf-pipeline');
+var path = require('path');
+var Promise = require('bluebird');
+var bufferToJson = require('../../lib/bufferToJson');
+var gzipTileset = require('../../lib/gzipTileset');
+var isGzippedFile = require('../../lib/isGzippedFile');
+var readFile = require('../../lib/readFile');
+var upgradeTileset = require('../../lib/upgradeTileset');
+
+var parseBinaryGltf = GltfPipeline.parseBinaryGltf;
+
+var batchedDeprecated1Directory = './specs/data/BatchedDeprecated1/';
+var batchedDeprecated2Directory = './specs/data/BatchedDeprecated2/';
+var gzippedDirectory = './specs/data/BatchedDeprecated-gzipped/';
+var upgradedDirectory = './specs/data/BatchedDeprecated-upgraded/';
+var upgradedJson = './specs/data/BatchedDeprecated-upgraded/tileset.json';
+var upgradedB3dm1 = './specs/data/BatchedDeprecated-upgraded/batchedDeprecated1.b3dm';
+var upgradedB3dm2 = './specs/data/BatchedDeprecated-upgraded/batchedDeprecated2.b3dm';
+
+describe('upgradeTileset', function() {
+    afterEach(function (done) {
+        Promise.all([
+            fsExtra.remove(upgradedDirectory),
+            fsExtra.remove(gzippedDirectory)
```

Comment:
since `upgradeTileset` takes a `write` callback, should we try to avoid filesystem changing operations in this spec?